### DISCUSSION TITLE
Problems with Celery and Redis

### DISCUSSION THREAD

**Discussion by kempert** [pretix/pretix#3421] [Q&A]
I have some Problems in my Pretix Instance the Worker Service fould be faild often.

Jun 19 23:34:11 pretix systemd[1]: Started pretix background worker.
Jun 19 23:34:27 pretix celery[120]:  
Jun 19 23:34:27 pretix celery[120]:  -------------- celery@pretix v5.2.7 (dawn-chorus)
Jun 19 23:34:27 pretix celery[120]: --- ***** -----
Jun 19 23:34:27 pretix celery[120]: -- ******* ---- Linux-5.15.107-2-pve-x86_64-with-glibc2.28 2023-06-19 21:34:27
Jun 19 23:34:27 pretix celery[120]: - *** --- * ---
Jun 19 23:34:27 pretix celery[120]: - ** ---------- [config]
Jun 19 23:34:27 pretix celery[120]: - ** ---------- .> app:         pretix:0x7f709522cf70
Jun 19 23:34:27 pretix celery[120]: - ** ---------- .> transport:   redis://127.0.0.1:6379/2
Jun 19 23:34:27 pretix celery[120]: - ** ---------- .> results:     redis://127.0.0.1/1
Jun 19 23:34:27 pretix celery[120]: - *** --- * --- .> concurrency: 1 (prefork)
Jun 19 23:34:27 pretix celery[120]: -- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
Jun 19 23:34:27 pretix celery[120]: --- ***** -----
Jun 19 23:34:27 pretix celery[120]:  -------------- [queues]
Jun 19 23:34:27 pretix celery[120]:                 .> background       exchange=default(direct) key=background.#
Jun 19 23:34:27 pretix celery[120]:                 .> checkout         exchange=default(direct) key=checkout.#
Jun 19 23:34:27 pretix celery[120]:                 .> default          exchange=default(direct) key=default.#
Jun 19 23:34:27 pretix celery[120]:                 .> mail             exchange=default(direct) key=mail.#
Jun 19 23:34:27 pretix celery[120]:                 .> notifications    exchange=default(direct) key=notifications.#
Jun 19 23:34:27 pretix celery[120]: [tasks]
Jun 19 23:34:27 pretix celery[120]:   . pretix.api.webhooks.manually_retry_all_calls
Jun 19 23:34:27 pretix celery[120]:   . pretix.api.webhooks.notify_webhooks
Jun 19 23:34:27 pretix celery[120]:   . pretix.api.webhooks.send_webhook
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.cancelevent.cancel_event
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.cart.add_items_to_cart
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.cart.apply_voucher
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.cart.clear_cart
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.cart.remove_cart_position
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.cart.set_cart_addons
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.currencies.fetch_cnb_cz_rates
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.currencies.fetch_ecb_rates
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.export.export
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.export.multiexport
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.export.scheduled_event_export
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.export.scheduled_organizer_export
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.invoices.invoice_pdf_task
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.mail.mail_send_task
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.notifications.notify
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.notifications.send_notification
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.orderimport.import_orders
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.orders.cancel_order
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.orders.perform_order
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.shredder.export
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.shredder.shred
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.tickets.generate
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.tickets.invalidate_cache
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.update_check.update_check
Jun 19 23:34:27 pretix celery[120]:   . pretix.base.services.waitinglist.assign_automatically
Jun 19 23:34:27 pretix celery[120]:   . pretix.control.views.checkin.CheckInListBulkActionView.async_execute
Jun 19 23:34:27 pretix celery[120]:   . pretix.control.views.subevents.SubEventBulkCreate.async_execute
Jun 19 23:34:27 pretix celery[120]:   . pretix.control.views.vouchers.VoucherBulkCreate.async_execute
Jun 19 23:34:27 pretix celery[120]:   . pretix.plugins.badges.tasks.badges_create_pdf
Jun 19 23:34:27 pretix celery[120]:   . pretix.plugins.banktransfer.tasks.process_banktransfers
Jun 19 23:34:27 pretix celery[120]:   . pretix.plugins.sendmail.tasks.send_mails_to_orders
Jun 19 23:34:27 pretix celery[120]:   . pretix.plugins.sendmail.tasks.send_mails_to_waitinglist
Jun 19 23:34:27 pretix celery[120]:   . pretix.plugins.stripe.tasks.stripe_verify_domain
Jun 19 23:34:27 pretix celery[120]:   . pretix.plugins.ticketoutputpdf.tasks.tickets_create_pdf
Jun 19 23:34:27 pretix celery[120]:   . pretix.presale.style.regenerate_css
Jun 19 23:34:27 pretix celery[120]:   . pretix.presale.style.regenerate_organizer_css
Jun 19 23:34:28 pretix celery[120]: [2023-06-19 21:34:28,050: INFO/MainProcess] Connected to redis://127.0.0.1:6379/2
Jun 19 23:34:28 pretix celery[120]: [2023-06-19 21:34:28,054: INFO/MainProcess] mingle: searching for neighbors
Jun 19 23:34:29 pretix celery[120]: [2023-06-19 21:34:29,063: INFO/MainProcess] mingle: all alone
Jun 19 23:34:29 pretix celery[120]: [2023-06-19 21:34:29,084: INFO/MainProcess] celery@pretix ready.
Jun 19 23:35:42 pretix celery[120]: [2023-06-19 21:35:42,080: WARNING/MainProcess] consumer: Connection to broker lost. Trying to re-establish the connection...
Jun 19 23:35:42 pretix celery[120]: Traceback (most recent call last):
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/celery/worker/consumer/consumer.py", line 332, in start
Jun 19 23:35:42 pretix celery[120]:     blueprint.start(self)
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/celery/bootsteps.py", line 116, in start
Jun 19 23:35:42 pretix celery[120]:     step.start(parent)
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/celery/worker/consumer/consumer.py", line 628, in start
Jun 19 23:35:42 pretix celery[120]:     c.loop(*c.loop_args())
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/celery/worker/loops.py", line 97, in asynloop
Jun 19 23:35:42 pretix celery[120]:     next(loop)
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/kombu/asynchronous/hub.py", line 362, in create_loop
Jun 19 23:35:42 pretix celery[120]:     cb(*cbargs)
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/kombu/transport/redis.py", line 1326, in on_readable
Jun 19 23:35:42 pretix celery[120]:     self.cycle.on_readable(fileno)
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/kombu/transport/redis.py", line 562, in on_readable
Jun 19 23:35:42 pretix celery[120]:     chan.handlers[type]()
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/kombu/transport/redis.py", line 906, in _receive
Jun 19 23:35:42 pretix celery[120]:     ret.append(self._receive_one(c))
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/kombu/transport/redis.py", line 916, in _receive_one
Jun 19 23:35:42 pretix celery[120]:     response = c.parse_response()
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/redis/client.py", line 1542, in parse_response
Jun 19 23:35:42 pretix celery[120]:     response = self._execute(conn, try_read)
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/redis/client.py", line 1518, in _execute
Jun 19 23:35:42 pretix celery[120]:     return conn.retry.call_with_retry(
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/redis/retry.py", line 49, in call_with_retry
Jun 19 23:35:42 pretix celery[120]:     fail(error)
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/redis/client.py", line 1520, in <lambda>
Jun 19 23:35:42 pretix celery[120]:     lambda error: self._disconnect_raise_connect(conn, error),
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/redis/client.py", line 1507, in _disconnect_raise_connect
Jun 19 23:35:42 pretix celery[120]:     raise error
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/redis/retry.py", line 46, in call_with_retry
Jun 19 23:35:42 pretix celery[120]:     return do()
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/redis/client.py", line 1519, in <lambda>
Jun 19 23:35:42 pretix celery[120]:     lambda: command(*args, **kwargs),
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/redis/client.py", line 1540, in try_read
Jun 19 23:35:42 pretix celery[120]:     return conn.read_response(disconnect_on_error=False)
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/redis/connection.py", line 874, in read_response
Jun 19 23:35:42 pretix celery[120]:     response = self._parser.read_response(disable_decoding=disable_decoding)
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/redis/connection.py", line 347, in read_response
Jun 19 23:35:42 pretix celery[120]:     result = self._read_response(disable_decoding=disable_decoding)
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/redis/connection.py", line 357, in _read_response
Jun 19 23:35:42 pretix celery[120]:     raw = self._buffer.readline()
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/redis/connection.py", line 260, in readline
Jun 19 23:35:42 pretix celery[120]:     self._read_from_socket()
Jun 19 23:35:42 pretix celery[120]:   File "/var/pretix/venv/lib/python3.9/site-packages/redis/connection.py", line 213, in _read_from_socket
Jun 19 23:35:42 pretix celery[120]:     raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
Jun 19 23:35:42 pretix celery[120]: redis.exceptions.ConnectionError: Connection closed by server.
Jun 19 23:35:42 pretix celery[120]: [2023-06-19 21:35:42,086: WARNING/MainProcess] /var/pretix/venv/lib/python3.9/site-packages/celery/worker/consumer/consumer.
Jun 19 23:35:42 pretix celery[120]: In Celery 5.1 we introduced an optional breaking change which
Jun 19 23:35:42 pretix celery[120]: on connection loss cancels all currently executed tasks with late acknowledgement enabled.
Jun 19 23:35:42 pretix celery[120]: These tasks cannot be acknowledged as the connection is gone, and the tasks are automatically redelivered back to the queue.
Jun 19 23:35:42 pretix celery[120]: You can enable this behavior using the worker_cancel_long_running_tasks_on_connection_loss setting.
Jun 19 23:35:42 pretix celery[120]: In Celery 5.1 it is set to False by default. The setting will be set to True by default in Celery 6.0.
Jun 19 23:35:42 pretix celery[120]:   warnings.warn(CANCEL_TASKS_BY_DEFAULT, CPendingDeprecationWarning)
Jun 19 23:35:42 pretix celery[120]: [2023-06-19 21:35:42,091: ERROR/MainProcess] consumer: Cannot connect to redis://127.0.0.1:6379/2: Error 111 connecting to 1
Jun 19 23:35:42 pretix celery[120]: Trying again in 2.00 seconds... (1/100)
Jun 19 23:35:44 pretix celery[120]: [2023-06-19 21:35:44,096: INFO/MainProcess] Connected to redis://127.0.0.1:6379/2
Jun 19 23:35:44 pretix celery[120]: [2023-06-19 21:35:44,098: INFO/MainProcess] mingle: searching for neighbors
Jun 19 23:35:45 pretix celery[120]: [2023-06-19 21:35:45,103: INFO/MainProcess] mingle: all alone

**I have installed the actual Pretix version. and Redis Version 6.0.8.**

739:M 19 Jun 2023 23:38:43.098 # User requested shutdown...
739:M 19 Jun 2023 23:38:43.098 * Saving the final RDB snapshot before exiting.
739:M 19 Jun 2023 23:38:43.106 * DB saved on disk
739:M 19 Jun 2023 23:38:43.106 * Removing the pid file.
739:M 19 Jun 2023 23:38:43.106 # Redis is now ready to exit, bye bye...
1023:C 19 Jun 2023 23:38:43.324 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1023:C 19 Jun 2023 23:38:43.324 # Redis version=6.0.8, bits=64, commit=00000000, modified=0, pid=1023, just started
1023:C 19 Jun 2023 23:38:43.324 # Configuration loaded

 Redis 6.0.8 (00000000/0) 64 bit
Running in standalone mode
Port: 6379
 PID: 1023

1023:M 19 Jun 2023 23:38:43.325 # Server initialized
1023:M 19 Jun 2023 23:38:43.325 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcom$
1023:M 19 Jun 2023 23:38:43.325 * Loading RDB produced by version 6.0.8
1023:M 19 Jun 2023 23:38:43.325 * RDB age 0 seconds
1023:M 19 Jun 2023 23:38:43.325 * RDB memory usage when created 1.16 Mb
1023:M 19 Jun 2023 23:38:43.327 * DB loaded from disk: 0.001 seconds
1023:M 19 Jun 2023 23:38:43.327 * Ready to accept connections

**Can anybody help me by the Problem? Pretix is shown a Rotating Gear wich sas processing and after a the a Internal Server error.**

**Comment by raphaelm** [comment]
The posted logs looks fine.

If pretix is showing an internal server error, there should be a more helpful log entry in pretix.log

**Reply by kempert** [reply]
At the Moments with the Log it wouldn't be fine, because of the worker can process anyway. I could solve IT by purge and new Install from redis-server.
